### PR TITLE
HIP updates

### PR DESCRIPTION
**Pull request by MichaelSt98**
* adding arch files for LUMI CCE 16.0.1
* adding col/s as metric for HIP variants

### CODE REVIEW

**Review by mlange05 — APPROVED** [review]
Changes look sane to me, but I'll let @reuterbal check the LUMI configuration before merging. GTG from me though

File: arch/eurohpc/lumi/cray-gpu/16.0.1/toolchain.cmake [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+set( ENABLE_OMP ON CACHE STRING "" )
+set( OpenMP_C_FLAGS   "-homp" CACHE STRING "" )
+set( OpenMP_Fortran_FLAGS   "-homp -hnoacc -hlist=aimd -maxrregcount 64" CACHE STRING "" )
+
+####################################################################
+# OpenACC FLAGS
+####################################################################
+
+set( ENABLE_ACC ON CACHE STRING "" )
```
Comment (reuterbal) [inline arch/eurohpc/lumi/cray-gpu/16.0.1/toolchain.cmake:28]: Isn't -maxrregcount 64 an NVHPC compiler option? It didn't work for me with CCE.
Also, there's the usual OpenMP library lookup failure that we know from IFS. The following options work, though:

```suggestion
set( OpenMP_C_FLAGS           "-fopenmp" )
set( OpenMP_CXX_FLAGS         "-fopenmp" )
set( OpenMP_Fortran_FLAGS     "-fopenmp" )
set( OpenMP_C_LIB_NAMES       "craymp" )
set( OpenMP_CXX_LIB_NAMES     "craymp" )
set( OpenMP_Fortran_LIB_NAMES "craymp" )
set( OpenMP_craymp_LIBRARY    "/opt/cray/pe/cce/16.0.1/cce/x86_64/lib/libcraymp.so" )

####################################################################
# OpenACC FLAGS
####################################################################

```

**Review by reuterbal — CHANGES_REQUESTED** [review]
Many thanks. HIP variants work perfectly as expected, and plain SCC still crashes the compiler, as expected.
But the OpenMP flags in the toolchain file aren't working for me as is. I've made a suggestion for a fix that I could build with successfully.

**Comment by reuterbal** [comment]
Thanks! I can confirm that all variants build now.

Unfortunately, the env file still doesn't quite work for me:
1. The import of cray-hdf5/1.12.2.3 is missing, which is required to build without Serialbox
2. somehow, none of the OpenACC or OpenMP offload variants seem to be offloading to GPU.

Would you mind fixing 1 and briefly investigating 2?

File: arch/eurohpc/lumi/cray-gpu/16.0.1/env.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+module_load craype-x86-trento
+module_load cray-mpich/8.1.27
+module_load craype-network-ofi
+module_load craype-accel-amd-gfx90a
+module_load cray-python/3.10.10
+module_load rocm/5.2.3
+module_load buildtools/22.12
+module_load LUMI/23.09
+module_load Boost/1.82.0-cpeCray-23.09
+module_load cray-hdf5/1.12.2.3
```
Comment (reuterbal) [inline arch/eurohpc/lumi/cray-gpu/16.0.1/env.sh:39]: Unfortunately, still a few things problematic with this:
* Loading LUMI/23.09 defaults back to `partition/L` (at least if building on the login node), thus unloading the craype-accel and craype-x86 modules)
* The HDF5 version is now 1.12.2.7 (not sure if that changed recently?)
* The buildtools version refers to an older LUST stack, thus the module is inactive (and we fall back to the default CMake 3.20)

For the record, this has worked for me as expected:
```suggestion
module_load LUMI/23.09
module_load partition/G
module_load PrgEnv-cray/8.4.0
module_load cce/16.0.1
module_load cray-mpich/8.1.27
module_load craype-network-ofi
module_load rocm/5.2.3
module_load buildtools/23.09
module_load Boost/1.82.0-cpeCray-23.09
module_load cray-python/3.10.10
module_load cray-hdf5/1.12.2.7
module_load craype-x86-trento
module_load craype-accel-amd-gfx90a
```

**Review by reuterbal — CHANGES_REQUESTED** [review]
Thanks for figuring out the missing offload. Still some problems with the env file, at least when I've tried it, but a suggested solution that worked for me.

**Comment by MichaelSt98** [comment]
Just double-checked your proposed changes and committed them.
Thank you @reuterbal!

**Review by reuterbal — APPROVED** [review]
Perfect, many thanks. Going in

**Pull request merged**